data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/SqsService.java

```diff
@@ -23,22 +24,27 @@ public class SqsService {
     private final S3Service s3Accessor;
     private final SqsClient sqsClient;
     private final PluginMetrics pluginMetrics;
+    private final boolean endToEndAcknowledgementsEnabled;
+    private final AcknowledgementSetManager acknowledgementSetManager;
 
     private Thread sqsWorkerThread;
 
-    public SqsService(final S3SourceConfig s3SourceConfig,
+    public SqsService(final AcknowledgementSetManager acknowledgementSetManager,
+                      final S3SourceConfig s3SourceConfig,
                       final S3Service s3Accessor,
                       final PluginMetrics pluginMetrics) {
         this.s3SourceConfig = s3SourceConfig;
         this.s3Accessor = s3Accessor;
         this.pluginMetrics = pluginMetrics;
+        this.acknowledgementSetManager = acknowledgementSetManager;
```

Comment:
nit: TIOLI, we already pass `s3SourceConfig` to `SqsWorker`, you don't need to get these values here. This will reduce the number of parameters passed to `SqsWorker`.

Reply:
Makes sense but only one parameter can be avoided, not both.